### PR TITLE
Bug fix 3.5/wait until we appear in supervision health

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Don't report to work until we have successfully entered ourselves in `Supervision/Health`.

- [x] Bug-Fix for a *released version* (did you remember to port this to all relevant release branches?)
- [x] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)
- [x] The behavior change can be verified via automatic tests

For the devel branch, the change is contained in a different branch (https://github.com/arangodb/arangodb/pull/10300)

### Testing & Verification

This change is already covered by existing tests, such as _any cluster test_.

https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/6876/

### CODE REVIEW

File: arangod/RestServer/BootstrapFeature.cpp [hunk 1]
```diff
@@ -331,6 +332,30 @@ void BootstrapFeature::start() {
...
+    LOG_TOPIC("4000c", DEBUG, arangodb::Logger::CLUSTER) << "waiting for our health entry to appear in Supervision/Health";
+    bool found = false;
+    AgencyComm agency;
+    int tries = 0;
+    while (++tries < 30) {
+      AgencyCommResult result = agency.getValues(::healthKey);
+      if (result.successful()) {
+        VPackSlice value = result.slice()[0].get(
+          std::vector<std::string>({AgencyCommManager::path(), "Supervision", "Health", ServerState::instance()->getId(), "Status"}));
+        if (value.isString() && !value.copyString().empty()) {
```
Comment (maierlars) [inline arangod/RestServer/BootstrapFeature.cpp]: `value.getStringLength() == 0`

Comment (jsteemann) [reply arangod/RestServer/BootstrapFeature.cpp]: Ok

Comment (jsteemann) [reply arangod/RestServer/BootstrapFeature.cpp]: Added that change

**Comment by jsteemann** [comment]
Tests blue

**Pull request merged**